src/transforms/transform-deployment.ts

```diff
@@ -155,6 +157,26 @@ export function mapEnvironment(environment: string): string {
 	return jiraEnv;
 }
 
+// Maps commit summaries to a commit association containing the commit keys (commit hash and repository id)
+function mapCommitSummariesToCommitAssociation(
+	commitSummaries: Array<CommitSummary>,
+	repositoryId: string
+): JiraCommitAssociation {
+
+	const commitKeys = commitSummaries
```

Comment:
if you're doing `commitSummaries?.` if means that commitSummaries can be undefined.  As such, the param needs to specify that with `commitSummaries?: Array<CommitSummary>`.  Also, if that's the case, it means that `commitKeys` can be undefined as well and the devinfo API will definitely fail as it wouldn't accept an association with missing values.

IMO, we've seen enough situations that it is possible that data might be missing, so we need to handle this edge case.  I think the function needs to return `undefined` if `commitSummaries` is undefined